runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectMetrics.java

```diff
@@ -217,19 +219,43 @@ public GaugeResult extract(GaugeData data) {
         }
       };
 
+  private static final MetricAggregation<StringSetData, StringSetResult> STRING_SET =
+      new MetricAggregation<StringSetData, StringSetResult>() {
+        @Override
+        public StringSetData zero() {
+          return StringSetData.empty();
+        }
+
+        @Override
+        public StringSetData combine(Iterable<StringSetData> updates) {
+          StringSetData result = StringSetData.empty();
+          for (StringSetData update : updates) {
+            result = result.combine(update);
```

Comment:
Doing a one-at-a-time binary combine of multiple sets might be inefficient (vs. creating a single set and adding everything to it). I don't know how often this method is called thoug.